### DISCUSSION TITLE
Is there a way to use regex for field columns in WHERE?

### DISCUSSION THREAD

**Discussion by aborruso** [duckdb/duckdb#6720] [Q&A]
Hi,
in SELECT I can use regex to select columns (`SELECT COLUMNS('number\d+') FROM addresses;`).

I have a table with more than 200 columns. I would like to run this kind of query

```
select  * from  input where COLUMNS('.+DATA.+') >'2022-10-10'
```

And to have in OR all the columns that match the regex.

Thank you

**Comment by Mytherin** [comment]
This is actually possible in the dev version, see #6621, but it does an `AND` of the conditions, not an `OR`.

**Reply by aborruso** [reply]
> This is actually possible in the dev version, see #6621, but it does an `AND` of the conditions, not an `OR`.

Wow, you are great.

Often I have the goal to find all the rows, in which I have a specific string in all the columns that have (in example) `DATE` in the field name. It would be very helpful to have `OR` as an option.

**Comment by soerenwolfers** [comment]
@aborruso It's now possible to do an `OR`, see https://github.com/duckdb/duckdb/issues/15676
